bindings/cpp/src/ethereum.rs

```diff
@@ -63,6 +65,18 @@ mod ffi {
             private_key: &PrivateKey,
         ) -> Result<Box<EthContract>>;
 
+        // extract toplevel json with keyname
+        fn extract_json(src: String, keyname: String) -> Result<String>;
+        // 0x... -> bytes
+        fn extract_bytes(src: String, keyname: String) -> Result<Vec<u8>>;
+
+        fn encode_deploy_contract(
+            rpcserver: String,
+            abi: String,
+            bytecode: Vec<u8>,
```

Comment:
These twos are actually from the abi file, how about make it just as simple as possible - read abi file by name, and do them in Rust, to replace std::ifstream to read file then extract_json, extract_bytes etc C++.